src/compositor/shells/xdg_v6.rs

```diff
@@ -17,6 +17,27 @@ impl XdgV6 {
 }
 
 impl XdgV6ShellHandler for XdgV6 {
+    fn resize_request(&mut self,
+                      compositor: CompositorHandle,
+                      _: SurfaceHandle,
+                      shell_surface: XdgV6ShellSurfaceHandle,
+                      event: &ResizeEvent) {
+        println!("got a resize event: {:?}", event);
+        with_handles!([(compositor: {compositor})] => {
+            let server: &mut Server = compositor.into();
+            let Server { ref mut seat,
+                         ref mut views,
+                         ref mut cursor,
+                         .. } = *server;
+            let resizing_shell = shell_surface.into();
+            let idx = views.iter().position(|view| view.shell == resizing_shell);
+            if let Some(idx) = idx {
+                let view = views.get(idx).unwrap().clone();
```

Comment:
Use `views[idx].clone()` here since you know the `idx` is within bounds.

In fact since you're not using the index for anything _really_ you can do this instead and you have guaranteed elided bounds checks if you do this instead:

```rust
for view in &views {
    if view.shell == resizing_shell {
        seat.begin_resize(cursor, view.clone(), views, event.edges());
        break;
    }
}
```

Reply:
Or actually event better, use [find](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find). Those functions usually generate better code than the iterative version:

```rust
if let Some(view) = views.iter().find(|view| view.shell == resizing_shell).cloned() {
    seat.begin_resize(cursor, view.clone(), views, event.edges())
}
```